Explorer/Assets/DCL/Multiplayer/SDK/Systems/SceneWorld/WritePlayerTransformSystem.cs

```diff
@@ -0,0 +1,61 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using CrdtEcsBridge.Components;
+using CrdtEcsBridge.Components.Transform;
+using CrdtEcsBridge.ECSToCRDTWriter;
+using DCL.Multiplayer.SDK.Components;
+using ECS.Abstract;
+using ECS.Groups;
+using ECS.LifeCycle.Components;
+using ECS.Prioritization.Components;
+using SceneRunner.Scene;
+using Utility;
+
+namespace DCL.Multiplayer.SDK.Systems.SceneWorld
+{
+    [UpdateInGroup(typeof(SyncedPreRenderingSystemGroup))]
+    // [UpdateBefore(typeof(CleanUpGroup))]
+    // [LogCategory(ReportCategory.PLAYER_AVATAR_BASE)]
+    public partial class WritePlayerTransformSystem : BaseUnityLoopSystem
+    {
+        private readonly IECSToCRDTWriter ecsToCRDTWriter;
+        private readonly ISceneData sceneData;
+
+        public WritePlayerTransformSystem(World world, IECSToCRDTWriter ecsToCRDTWriter, ISceneData sceneData) : base(world)
+        {
+            this.ecsToCRDTWriter = ecsToCRDTWriter;
+            this.sceneData = sceneData;
+        }
+
+        protected override void Update(float t)
+        {
+            HandleComponentRemovalQuery(World);
+            UpdateSDKTransformQuery(World);
+        }
+
+        [Query]
+        [None(typeof(DeleteEntityIntention))]
+        private void UpdateSDKTransform(ref PlayerCRDTEntity playerCRDTEntity, ref SDKTransform sdkTransform)
+        {
+            // Main player Transform is handled by 'WriteMainPlayerTransformSystem'
+            if (playerCRDTEntity.CRDTEntity.Id == SpecialEntitiesID.PLAYER_ENTITY) return;
+
+            // Patch position to be scene-relative
+            sdkTransform.Position = ParcelMathHelper.GetSceneRelativePosition(sdkTransform.Position, sceneData.Geometry.BaseParcelPosition);
+
+            ecsToCRDTWriter.PutMessage<SDKTransform, SDKTransform>(static (pbComponent, transform) =>
```

Comment:
Can't you use `ExposedTransformUtils` instead?

Reply:
Ah good call internally `ExposedTransformUtils.Put()` already does the same scene-relative position conversion that I was doing in there.

I made `SDKTransform` implement `IExposedTransform` so that it can be sent through `ExposedTransformUtils.Put()`, that forced me to change several other usages of the `SDKTransform.Position` and `Rotation` properties, I implemented the needed changes at https://github.com/decentraland/unity-explorer/pull/1030/commits/b5c6df3d83c363bd98c1a924b75d0c43faf05742 and tested that the feature keeps working as expected.

Reply:
I'm not sure if making `SDKTransform` implement `IExposedTransform` may impact performance?

Maybe it's better to revert that change and just implement an overload for [ExposedTransformUtils.Put](https://github.com/decentraland/unity-explorer/blob/b5c6df3d83c363bd98c1a924b75d0c43faf05742/Explorer/Assets/Scripts/ECS/Unity/Transforms/ExposedTransformUtils.cs#L11) allowing the use of just `Transform` instead of `IExposedTransform` ?

Reply:
No, it's not gonna impact as `SDKTrasnform` is a class, not a structure, otherwise I would do a generic with `IExposedTransform` contraint